src/taskgraph/transforms/chunking.py

```diff
@@ -0,0 +1,94 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+import copy
+
+from voluptuous import ALLOW_EXTRA, Optional, Required
+
+from taskgraph.transforms.base import TransformSequence
+from taskgraph.util.schema import Schema
+from taskgraph.util.substitution import substitute
+
+CHUNK_SCHEMA = Schema(
+    {
+        # Optional, so it can be used for a subset of jobs in a kind
+        Optional("chunk-config"): {
+            Required("total-chunks"): int,
+            Optional("substitution-fields"): [str],
+        }
+    },
+    extra=ALLOW_EXTRA,
+)
+
+chunk = TransformSequence()
```

Comment:
Keep in mind that there's logic that defaults this variable to `transforms`. I think I'd keep that default for the `chunk` case here (I imagine `unchunk` will be much less common).

Reply:
Good point, will do.